src/ServiceManagement/RecoveryServices/Commands.RecoveryServices/Service/InvokeAzureRecoveryServicesVaultUpgrade.cs

```diff
@@ -0,0 +1,238 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Management.Automation;
+using System.Threading;
+using Microsoft.Azure.Commands.RecoveryServices.SiteRecovery;
+using Microsoft.WindowsAzure.Management.RecoveryServicesVaultUpgrade.Models;
+
+namespace Microsoft.Azure.Commands.RecoveryServices
+{
+    /// <summary>
+    /// Used to initiate vault upgrade operation.
+    /// </summary>
+    [Cmdlet(VerbsLifecycle.Invoke, "AzureRecoveryServicesVaultUpgrade", SupportsShouldProcess = true)]
+    public class InvokeAzureRecoveryServicesVaultUpgrade : RecoveryServicesCmdletBase
+    {
+        #region Parameters
+
+        /// <summary>
+        /// Gets or sets vault type.
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateSet(
+            Constants.ASRVaultType,
+            Constants.BackupVault,
+            IgnoreCase = true)]
+        public string ResourceType { get; set; }
+
+        /// <summary>
+        /// Gets or sets vault name.
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateNotNullOrEmpty]
+        public string VaultName { get; set; }
+
+        /// <summary>
+        /// Gets or sets location of the vault.
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        /// <summary>
+        /// Gets or sets target resource group.
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateNotNullOrEmpty]
+        [Alias("TargetRG", "TargetRGName", "RG")]
+        public string TargetResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets switch parameter. On passing, command does not ask for confirmation.
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter Force { get; set; }
```

Comment:
a) I copy pasted the exact code: if (ShouldProcess(target, message) && (!Force.IsPresent || ShouldContinue(query, caption))) { ... cmdlet execution }

In my scenario, the expected behavior is with force, it shouldn't ask for confirmations. Without force, it should.
Whereas the opposite was happening here: with force, it was asking for confirmation. 

Also since anyways we want to expose force parameter, can i continue with the approach I am using currently?